fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -1377,15 +1400,46 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 		return res;
 	}
 
-	Future<Void> addShard(KeyRangeRef range, UID id) override {
+	void addShard(KeyRangeRef range, UID id) override {
 		if (!SERVER_KNOBS->ROCKSDB_ENABLE_SHARDING) {
 			// TODO: add a new error code.
 			return internal_error();
 		}
-		auto a = std::make_unique<Writer::AddShardAction>(dataPath, id, range, &shardMap);
-		Future<Void> f = a->done.getFuture();
-		writeThread->post(a.release());
-		return f;
+
+		if (dirtyShards == nullptr) {
+			dirtyShards.reset(new std::set<std::shared_ptr<DataShard>>());
```

Comment:
If we delay the creation of Rocks Instance to the first write, we don't need to mark it as dirty here.